### PR TITLE
Fix GPTQ for inputs with batch size != 1 and with seq len == 1

### PR DESCRIPTION
**Pull request by ljaljushkin**
### Changes

GPTQ correctly processes inputs with batch size != 1 and with batch size and sequence length equal 1.
Also changed the errors we are raising in NNCF from built-in Python errors to NNCF-specific ones.

### Reason for changes

Stable-diffusion models, e.g. `runwayml/stable-diffusion-v1-5` has as an input for linear layers with the following shapes: `[2*num_images_in_prompt, text_embedding_size, hidden_dimension]`. 

https://github.com/openvinotoolkit/nncf/blob/develop/examples/llm_compression/openvino/tiny_llama/main.py
uses not filtered data from `wikitext` that leads to the corner case with sequence length == 1.

### Related tickets

150851, 155538

### Tests

- test_compression_with_transposed_activations
- test_compression_with_different_algo_combinations
- test_raise_error_with_unsupported_params_for_e2m1
- test_raise_error_with_unsupported_params_for_empty_dataset

CI

- [ ] weight compression conformance

### CODE REVIEW

File: nncf/quantization/algorithms/weight_compression/activation_stats.py [hunk 1]
```diff
@@ -28,7 +28,10 @@ def process_stats(stats: List[Tensor], subset_size: int) -> Tuple[Tensor, Tensor
         X - average channel magnitude across tokens in the sequence [HiddenDim, SampleSize]
     :rtype: Tuple[TTensor, TTensor]
     """
-    X = fns.stack([fns.mean(stat, axis=0) for stat in stats])  # [Batch, HiddenDim]
+    # Transposed input activations are not supported, the hidden dimension is always the last dimension.
+    # Need to reduce over all axis, except hidden dimension.
+    reduction_axis = tuple(range(stats[0].ndim)[:-1])
+    X = fns.stack([fns.mean(stat, axis=reduction_axis) for stat in stats])  # [Batch, HiddenDim]
```
Comment (ljaljushkin) [inline nncf/quantization/algorithms/weight_compression/activation_stats.py:34]: Fix for standalone data-aware methods

File: nncf/quantization/algorithms/weight_compression/gptq.py [hunk 2]
```diff
@@ -264,7 +264,7 @@ def _quantize_weights(
                         scales.append(scale)
                     else:
                         if self._scale_estimation and block_compression_config.num_bits == 4:
-                            activations = [inp.squeeze()[:, (i1 + i) : (i1 + i + group_size)] for inp in inputs]
+                            activations = [inp.squeeze()[..., (i1 + i) : (i1 + i + group_size)] for inp in inputs]
```
Comment (ljaljushkin) [inline nncf/quantization/algorithms/weight_compression/gptq.py:267]: fix for GPTQ + Scale Estimation

**Review by alexsu52 — APPROVED** [review]
LGTM

File: nncf/quantization/algorithms/weight_compression/algorithm.py [hunk 3]
```diff
@@ -369,7 +369,7 @@ def apply(
         self._set_weight_compression_config(ratio_defining_params, model, graph, activations)
         nncf_logger.info(self._get_bitwidth_distribution_str(all_weight_params, ratio_defining_params))
 
-        if self._awq and activations is not None and self._mode != CompressWeightsMode.E2M1:
```
[the remark this answers is not in the document]
Comment (ljaljushkin) [reply nncf/quantization/algorithms/weight_compression/algorithm.py:372]: Because it's already checked here: https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/quantize_model.py#L518-L524
Unit tests check that:
https://github.com/openvinotoolkit/nncf/pull/3002/files#diff-223ea638f7751f7c0c3e8f867ec9c8c132a3ccd62a9dcea2a5d158836c71c222R730-R741

**Comment by kshpv** [comment]
This PR resolves 155538

**Review by andreyanufr — APPROVED** [review]

File: nncf/quantization/algorithms/weight_compression/gptq.py [hunk 4]
```diff
@@ -264,7 +264,7 @@ def _quantize_weights(
                         scales.append(scale)
                     else:
                         if self._scale_estimation and block_compression_config.num_bits == 4:
-                            activations = [inp.squeeze()[:, (i1 + i) : (i1 + i + group_size)] for inp in inputs]
+                            activations = [inp[..., (i1 + i) : (i1 + i + group_size)] for inp in inputs]
```
Comment (ljaljushkin) [inline nncf/quantization/algorithms/weight_compression/gptq.py:267]: Slice by last dimension, which is supposed to be hidden one.
It's aligned with processing statistics in `activations_to_wc_statistics` when reduction axes are all dimensions except the last one

File: tests/openvino/native/quantization/test_weights_compression.py [hunk 5]
```diff
@@ -1383,6 +1420,14 @@ def test_data_aware_algo_with_different_activation_dimensions(n_extra_dims):
     )
 
 
+@pytest.mark.parametrize(
+    "input_shape",
+    [
+        LMLinearModel.INPUT_SHAPE,
+        [3, 5, 16],
+        [1, 1, 16],
```
Comment (ljaljushkin) [inline tests/openvino/native/quantization/test_weights_compression.py:1428]: Added test case for the bug 155538 with tiny-llama example. The root cause is not filtered data with sequence length == 1.
@kshpv

File: tests/openvino/native/quantization/test_weights_compression.py [hunk 6]
```diff
@@ -1383,6 +1420,14 @@ def test_data_aware_algo_with_different_activation_dimensions(n_extra_dims):
     )
 
 
+@pytest.mark.parametrize(
+    "input_shape",
+    [
+        LMLinearModel.INPUT_SHAPE,
+        [3, 5, 16],
```
Comment (ljaljushkin) [inline tests/openvino/native/quantization/test_weights_compression.py:1427]: test case to cover SD case with batch size != 1

**Review by nikita-savelyevv — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]
